src/Mvc/Mvc/src/MvcServiceCollectionExtensions.cs

```diff
@@ -253,7 +253,7 @@ private static IMvcCoreBuilder AddControllersWithViewsCore(IServiceCollection se
     /// To add services for controllers with views call <see cref="AddControllersWithViews(IServiceCollection)"/>.
     /// </para>
     /// </remarks>
-    [RequiresUnreferencedCode("Razor Pages does not currently support native AOT.", Url = "https://aka.ms/aspnet/nativeaot")]
+    [RequiresUnreferencedCode("Razor Pages does not currently support trimming or native AOT,", Url = "https://aka.ms/aspnet/trimming")]
```

Comment:
Do we need this change in other messages that were added in #47663?

Reply:
I think the nuance here was that razor pages also didn't support trimming. That isn't always the case in other areas.

Reply:
If it isn't the case in the other areas, we should be using `RequiresUnreferencedCode`. If they support trimming, but don't support NativeAOT - we should be using `RequiresDynamicCode` attributes.

Reply:
TBH I haven't looked that closely. This change was made by specific request because I think there was some confusion. I'm happy to go through and update all of the changes we made for #47663 if the clarification is required everywhere.

Reply:
All the attributes we added have this same issue. If you just "PublishTrimmed=true" an ASP.NET app calling those APIs, you will get a warning saying:

```
XXX API does not currently support native AOT.
```